activemq-broker/src/main/java/org/apache/activemq/broker/TransportConnection.java

```diff
@@ -196,9 +190,10 @@ public void onCommand(Object o) {
                         throw new RuntimeException("Protocol violation - Command corrupted: " + o.toString());
                     }
                     Command command = (Command) o;
-                    if (!brokerService.isStopping()) {
+                    final int[] processDuringShutdown = { CommandTypes.SHUTDOWN_INFO, CommandTypes.REMOVE_INFO };
+                    if (!brokerService.isStopping() || Arrays.asList(processDuringShutdown).contains(command.getDataStructureType())) {
                         Response response = service(command);
-                        if (response != null && !brokerService.isStopping()) {
+                        if (response != null && Arrays.asList(processDuringShutdown).contains(command.getDataStructureType())) {
```

Comment:
This bit looks to filter any response that isn't a ShutdownInfo or a RemoveInfo response which I don't think you would want to do.